### PR TITLE
Add support for Django 2.0

### LINKED ISSUE #1268: Migrate website to Django 2.0
Django 2.0 was released few months ago, and the python.org website should be migrated to it. The biggest challenge is switching to the new URL dispatcher.

But the release notes should contain information on what needs to be migrated: https://docs.djangoproject.com/en/2.0/releases/2.0/

Also, running the automatic tests and performing manual testing should help identify what needs to be changed.

### PR DESCRIPTION
**Pull request by antdking**
fixes #1268

### CODE REVIEW

File: pydotorg/urls.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
 handler404 = custom_404
 
+
```
Comment (berkerpeksag) [inline pydotorg/urls.py]: Style nit: Did you get a warning from your choice of linter or is this just an oversight?

File: pydotorg/resources.py [hunk 2]
```diff
@@ -120,3 +120,4 @@ class Meta:
         authentication = ApiKeyOrGuestAuthentication()
         authorization = StaffAuthorization()
         throttle = CacheThrottle(throttle_at=600) # default is 150 req/hr
+        abstract = True
```
Comment (berkerpeksag) [inline pydotorg/resources.py:123]: Did you get any error without this line? I couldn't find anything related at https://github.com/django-tastypie/django-tastypie/blob/master/docs/release_notes/v0.14.0.rst Or did something changed in Django 2.0?

**Review by berkerpeksag — COMMENTED** [review]
Looks good, thank you! Do you have a plan for switching to use the new `path()` API or do you want leave it to another PR?

File: pydotorg/urls.py (hunk 1 above)
Comment (antdking) [reply pydotorg/urls.py]: Oversight

File: pydotorg/resources.py (hunk 2 above)
Comment (antdking) [reply pydotorg/resources.py:123]: Unfortunately, tp 0.14.1 doesn't have release notes, but is needed for django 2 support.

The need for this is from this commit: https://github.com/django-tastypie/django-tastypie/commit/7dedb7be80868ee5dc72b79226b89bdb66f8a172

A workaround was added here: https://github.com/django-tastypie/django-tastypie/pull/1519

Not ideal, as child classes must be explicitly abstract False, however that's what has been provided

**Comment by antdking** [comment]
The new `path()` syntax can be another PR. It's not required to get dj2 working

File: pydotorg/resources.py (hunk 2 above)
Comment (berkerpeksag) [reply pydotorg/resources.py:123]: Thanks. We are going to remove tastypie soon so we can live with bunch of `abstract = False` additions :)

**Comment by antdking** [comment]
@berkerpeksag, whitespace issues addressed

**Review by berkerpeksag — APPROVED** [review]

**Comment by berkerpeksag** [comment]
@cybojenix LGTM, thank you! Just one more question: Do you have any team members I need to give credit in the commit message?

**Comment by antdking** [comment]
@berkerpeksag, yes. @dcreekp, and I can't remember the other unfortunately. @dcreekp may know though

**Comment by berkerpeksag** [comment]
@dcreekp could you please give your and the other contributor's name in the following format?

```
Your Name <your.name@example.com>
```

If you don't want to share your public email address, you can follow the guide at https://help.github.com/articles/creating-a-commit-with-multiple-authors/#required-co-author-information

**Comment by dcreekp** [comment]
@berkerpeksag @cybojenix 
here are my details:
Tarbo Fukazawa <dcreekp@users.noreply.github.com>
thanks.
and I've just chased up Paul (I don't know his github name)

**Comment by prdickson** [comment]
@berkerpeksag @cybojenix 
here are my details...
Paul Dickson prdickson@gmail.com

**Comment by berkerpeksag** [comment]
@cybojenix @dcreekp @prdickson thank you! This should be online at staging.python.org in an hour.

**Pull request merged**